components/calendar/benches/date.rs

```diff
@@ -0,0 +1,159 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+mod fixtures;
+
+use crate::fixtures::structs::DateFixture;
+use criterion::{
+    black_box, criterion_group, criterion_main, measurement::WallTime, BenchmarkGroup, Criterion,
+};
+use icu_calendar::{
+    iso::IsoDay, iso::IsoMonth, iso::IsoYear, AsCalendar, Calendar, Date, DateDuration,
+};
+
+fn bench_dates<A: AsCalendar>(dates: &mut [Date<A>]) {
```

Comment:
Suggestion (optional): actually, even cleaner would be to just take a `mut Date<A>` and call the function two times.

Reply:
I was debating between the two styles -- will go with calling the function twice